### DISCUSSION TITLE
Fastddsgen stucks when generating type definition files

### DISCUSSION THREAD

**Discussion by rookybird** [eProsima/Fast-DDS-Gen#175] [Q&A] [answered]
[idl.zip](https://github.com/eProsima/Fast-DDS/files/11458648/idl.zip)
RNDFMap includes GlobalPath.idl which includes Waypoint.idl
When I do "fastddsgen RNDFMap.idl", my computer stucks there for a long time. I am wondering what triggered this problem.
I am using fastddsgen 2.4.0

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution @rookybird. I am transferring this discussion to the Fast DDS-Gen repository as it is not really related with Fast DDS library.

**Comment by richiware** [comment, marked as answer]
We were testing your case and we found what is happening. It's caused by the calculation of the max serialized size of a type. Prior to 2.4.0, fastddsgen generates C++ code which realized the calculation in the initialization of the user's application. With 2.4.0, the calculation is done by fastddsgen when generating the source code.

In your case, you have nested sequences (several level of nesting), without boundaries (default 100 elements). Taking into account java code execution is slower than C++, the calculation takes for your case a lot of time, but finally fastddsgen finishes.

We take a note to improve this algorithm. A quick solution until then is, if you can, set boundaries in your sequences: `sequence<int, 30>`, to make faster the calculation.

**Comment by richiware** [comment]
Current `master` branch improve the max serialized size calculation. With the provided IDL files, current `fastddsgen` generates the source code in less 3 seconds.
